### PR TITLE
[SPARK-27946][SQL][FOLLOW-UP] Change doc and error message for SHOW CREATE TABLE

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This is a follow-up for #24938 to tweak error message and migration doc.

### Why are the changes needed?

Making user know workaround if SHOW CREATE TABLE doesn't work for some Hive tables.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Existing unit tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @gatorsmile

**Comment by viirya** [comment]
retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala [hunk 1]
```diff
@@ -1076,7 +1076,9 @@ case class ShowCreateTableCommand(table: TableIdentifier)
             "Failed to execute SHOW CREATE TABLE against table " +
               s"${tableMetadata.identifier}, which is created by Hive and uses the " +
               "following unsupported feature(s)\n" +
-              tableMetadata.unsupportedFeatures.map(" - " + _).mkString("\n")
+              tableMetadata.unsupportedFeatures.map(" - " + _).mkString("\n") + ". " +
+              s"Please use `SHOW CREATE TABLE ${tableMetadata.identifier} AS SERDE` " +
+              "to show Hive DDL instead."
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Do we have any test for this error case?

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Yea, test case `hive partitioned view is not supported` in `HiveShowCreateTableSuite` is for this error.

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Oh, I see. Thanks.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by viirya** [comment]
I'm going to merge this to master and branch-3.0.

**Pull request closed without merging**